mods/tnt/init.lua

```diff
@@ -384,7 +349,73 @@ minetest.register_craft({
 	}
 })
 
+function tnt.register_tnt(def)
+	local name = ""
+	if not def.name:find(':') then
+		name = "tnt:"..def.name
+	else
+		name = def.name
+		def.name = def.name:sub(def.name:find(':')+1)
+	end
+	if not def.tiles then def.tiles = {} end
+	local tnt_top = def.tiles.top or def.name.."_top.png"
+	local tnt_bottom = def.tiles.bottom or def.name.."_bottom.png"
+	local tnt_side = def.tiles.side or def.name.."_side.png"
+	local tnt_burning = def.tiles.burning or def.name.."_top_burning_animated.png"
+	if not def.damage_radius then 
+		def.damage_radius = def.radius * 2 
+	end
+	minetest.register_node(":"..name, {
+	description = def.description,
+	tiles = {tnt_top, tnt_bottom, tnt_side},
+	is_ground_content = false,
+	groups = {dig_immediate=2, mesecon=2, tnt=1},
+	sounds = default.node_sound_wood_defaults(),
+	on_punch = function(pos, node, puncher)
+		if puncher:get_wielded_item():get_name() == "default:torch" then
+			minetest.sound_play("tnt_ignite", {pos=pos})
+			minetest.set_node(pos, {name=name.."_burning"})
+			minetest.get_node_timer(pos):start(4)
+		end
+	end,
+	on_blast = function(pos, intensity)
+		tnt.burn(pos)
+	end,
+	mesecons = {effector = {action_on = 
+	function (pos)
+		tnt.boom(pos, def.radius, def.damage_radius, def.disable_drops)
+	end
+	}},
+})
```

Comment:
the problem is that the function is on the same indent as mesecons, it should be place one indent deeper.

```lua
	mesecons = {effector = {action_on = 
 		function (pos)
 			tnt.boom(pos, def.radius, def.damage_radius, def.disable_drops)
	 	end
	}},
```